c7n/resources/ecs.py

```diff
@@ -247,6 +248,37 @@ class ServiceTaskDefinitionFilter(RelatedTaskDefinitionFilter):
     """
 
 
+@Service.action_registry.register('modify')
+class UpdateService(BaseAction):
+    """Update service(s)."""
+
+    permissions = ('ecs:UpdateService',)
+
+    def validate(self):
+        api_call_shape = self.data.get('modify')
+        if not api_call_shape:
+            return
```

Comment:
you shouldn't need to validate this here, just use the schema validation (which you're missing)